enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1836,6 +1862,63 @@ func (p *PebbleCache) refreshMetrics(quitChan chan struct{}) {
 	}
 }
 
+func (p *PebbleCache) SupportsCompressor(compressor repb.Compressor_Value) bool {
+	switch compressor {
+	case repb.Compressor_IDENTITY:
+		return true
+	case repb.Compressor_ZSTD:
+		return p.enableZstdCompression
+	default:
+		return false
+	}
+}
+
+// compressionReader helps manage resources associated with a compression.NewZstdCompressingReader
+type compressionReader struct {
+	io.ReadCloser
+	readBuf     []byte
+	compressBuf []byte
+	bufferPool  *bytebufferpool.Pool
+}
+
+func (r *compressionReader) Close() error {
+	err := r.ReadCloser.Close()
+	r.bufferPool.Put(r.readBuf)
+	r.bufferPool.Put(r.compressBuf)
+	return err
+}
+
+func (p *PebbleCache) readerForCompressionType(reader io.ReadCloser, resource *resource.ResourceName, requestedCompression repb.Compressor_Value, cachedCompression repb.Compressor_Value) (io.ReadCloser, error) {
+	if requestedCompression != cachedCompression {
```

Comment:
You can reduce indentation by performing the inverse check and returning early.

Reply:
Done